src/Microsoft.OData.Edm/Csdl/Parsing/Common/EdmXmlDocumentParser.cs

```diff
@@ -407,40 +407,64 @@ protected string ValidateEnumMembersPath(string path)
 
         private string ValidateTypeName(string name)
         {
-            string[] typeInformation = name.Split(new char[] { '(', ')' });
-            string typeName = typeInformation[0];
+            Debug.Assert(name != null, "name != null");
+            ReadOnlySpan<char> nameSpan = name.AsSpan();
 
-            // For inline types, we need to check that the name contained inside is a valid type name
-            switch (typeName)
+            int indexOfOpenParen = nameSpan.IndexOf('(');

```

Comment:
how about the nested parenthesis?

Reply:
@xuzhg Can you share an example? I can add a test for that

Reply:
"collection(ns.Type(anything)))
"((ns.Type)"
"ns.()()"